test/parallel/test-tls-parse-cert-string.js

```diff
@@ -33,6 +41,26 @@ const tls = require('tls');
 
 {
   const invalid = 'fhqwhgads';
-  const invalidOut = tls.parseCertString(invalid);
+  const invalidOut = internalTLS.parseCertString(invalid);
   assert.deepStrictEqual(invalidOut, {});
 }
+
+common.restoreStderr();
+
+{
+  const regexp = new RegExp('^\\(node:\\d+\\) \\[DEP00XX\\] Deprecation' +
+                            'Warning: tls\\.parseCertString is deprecated\\n$');
+
+  const hijackTest = common.mustCall(function(data) {
+    assert.ok(regexp.test(data));
+  });
+
+  // test for deprecate message
+  common.hijackStderr(function(data) {
+    common.restoreStderr();
+    process.nextTick(hijackTest.bind(null, data));
+  });
```

Comment:
Please just use `common.expectWarning()` to detect the deprecation warning. There's no need to hijack the stderr.